support-workers/src/main/scala/com/gu/support/workers/lambdas/SendAcquisitionEvent.scala

```diff
@@ -54,6 +63,13 @@ object SendAcquisitionEvent {
     override def getMessage: String = errors.map(_.getMessage).mkString(". ")
   }
 
+  def paymentProviderFromPaymentMethod(paymentMethod: PaymentMethod): thrift.PaymentProvider =
+    paymentMethod match {
+      case _: CreditCardReferenceTransaction => thrift.PaymentProvider.Stripe
+      case _: PayPalReferenceTransaction => thrift.PaymentProvider.Paypal
+      case _: DirectDebitPaymentMethod | _: ClonedDirectDebitPaymentMethod => thrift.PaymentProvider.Gocardless
```

Comment:
what would happen if we got a new payment method?  Probably would want to add an alarm even if it were no payment for 24 hours.  I'm not sure if the code would throw an error if we missed one from the pattern match.

Reply:
yes `PaymentMethod` is a sealed trait so the compiler would warn.
Also, I haven't actually changed this code, just moved it out of a function